core/include/mutex.h

```diff
@@ -116,6 +120,13 @@ static inline void mutex_lock(mutex_t *mutex)
 /**
  * @brief Unlocks the mutex.
  *
+ * If there is more than one thread waiting for the mutex, the first one that
+ * requested the mutex will be unlocked (i.e. first come, first serve).
+ *
+ * In contrast with other systems, a mutex in RIOT can be unlocked by a thread
```

Comment:
How about just "A mutex can be unlocked by any thread or ISR."
There are "other systems" which behave simliar, e.g., Pythons Lock().